packages/typespec-vscode/src/vscode-cmd/openapi3-preview.ts

```diff
@@ -0,0 +1,323 @@
+              </head>
+              <body class="no-margin-padding full-device-view" style="background-color:white">
+                <div id="swagger-ui"></div>
+                <script src="${bundleJs}" charset="UTF-8"> </script>
+                <script src="${presetJs}" charset="UTF-8"> </script>
+                <script src="${initJs}" charset="UTF-8"> </script>
+              </body>
+            </html>`;
+  }
+
+  panel.webview.html = html;
+}
+
+async function getOutputFolder(mainTspFile: string): Promise<string | undefined> {
+  let tmpFolder = openApi3TempFolders.get(mainTspFile);
+  if (!tmpFolder) {
+    tmpFolder = await createTempDir();
+    if (tmpFolder) {
+      openApi3TempFolders.set(mainTspFile, tmpFolder);
+    }
+  }
+  return tmpFolder;
+}
+
+export async function clearOpenApi3PreviewTempFolders() {
+  for (const folder of openApi3TempFolders.values()) {
+    try {
+      await rm(folder, { recursive: true, force: true });
+    } catch (e) {
+      logger.error(`Failed to delete temporary folder: ${folder}`, [e]);
+    }
+  }
+}
+
+async function selectAndGetOpenApi3Content(
+  mainTspFile: string,
+  outputFolder: string,
+  selectOutput: boolean,
+  context: vscode.ExtensionContext,
+): Promise<string | undefined> {
+  let outputs: string[] | undefined;
+  try {
+    outputs = await readdir(outputFolder);
+  } catch (e) {
+    logger.error(`Failed to read output folder: ${outputFolder}`, [e], {
+      showOutput: true,
+      showPopup: true,
+    });
+    return;
+  }
+
+  if (outputs.length === 0) {
+    logger.error("No openAPI3 file generated.", [], {
+      showOutput: true,
+      showPopup: true,
+    });
+    return;
+  } else if (outputs.length === 1) {
+    const first = outputs[0];
+    const filePath = joinPaths(outputFolder, first);
+    selectedOpenApi3OutputFiles.set(mainTspFile, filePath);
+    return parseOpenApi3File(filePath);
+  } else {
+    if (selectedOpenApi3OutputFiles.has(mainTspFile) && !selectOutput) {
+      return parseOpenApi3File(selectedOpenApi3OutputFiles.get(mainTspFile)!);
+    }
+    const files = outputs.map<vscode.QuickPickItem & { path: string }>((file) => {
+      const filePath = joinPaths(outputFolder, file);
+      return {
+        label: getBaseFileName(file),
+        detail: filePath,
+        path: filePath,
+        iconPath: {
+          light: vscode.Uri.file(context.asAbsolutePath(`./icons/openapi3.light.svg`)),
+          dark: vscode.Uri.file(context.asAbsolutePath(`./icons/openapi3.dark.svg`)),
+        },
+      };
+    });
+    const selected = await vscode.window.showQuickPick(files, {
+      title: TITLE,
+      placeHolder: "Multiple OpenAPI3 files found. Select one to preview",
+    });
+    if (selected) {
+      selectedOpenApi3OutputFiles.set(mainTspFile, selected.path);
+      return parseOpenApi3File(selected.path);
+    } else {
+      logger.info("No OpenAPI3 file selected", [], {
+        showOutput: true,
```

Comment:
it's user's explicit choice, no need to showOutput/popup.

Reply:
changed according to comments